src/compiler/checker.rs

```diff
@@ -880,12 +1027,10 @@ impl<'type_checker> CheckTypeRelatedTo<'type_checker> {
         result != Ternary::False
     }
 
-    fn report_error(&self, message: &DiagnosticMessage) {
+    fn report_error(&self, message: &DiagnosticMessage, args: Option<Vec<String>>) {
         Debug_.assert(self.error_node.is_some(), None);
-        self.set_error_info(chain_diagnostic_messages(
-            self.error_info().clone(),
-            message,
-        ));
+        let error_info = { chain_diagnostic_messages(self.error_info().clone(), message, args) };
```

Comment:
Added an artificial block scope here to avoid conflicting borrows of `self.error_info`